### PR TITLE
Feature/cursor batch

### PR DESCRIPTION
**Pull request by JamesSCBD**
Add `vue-async-computed` for async method. So is it ok or find another solution

### CODE REVIEW

File: components/MultiDeleteModal.vue [hunk 1]
```diff
@@ -133,38 +146,50 @@ export default {
...
+      while (obj = await cursor.next()) { // eslint-disable-line no-cond-assign, no-await-in-loop
+        list.push({
+          name          : getName(obj).trim(),
+          organization  : getOrganization.call(this, obj).trim(),
+          emailOrCountry: getEmailOrCountry.call(this, obj).trim(),
+        });
+      }
+      return list;
+    },
+  },
```
[the remark this answers is not in the document]
Comment (sbilodeau) [reply components/MultiDeleteModal.vue:192]: I added the skip option to getCursor() as described

File: package.json [hunk 2]
```diff
@@ -30,6 +30,7 @@
     "luxon": "1.21.3",
     "nuxt": "2.11.0",
     "nuxt-fontawesome": "0.4.0",
+    "vue-async-computed": "^3.8.2",
```
Comment (sbilodeau) [inline package.json:33]: based on my comment at https://github.com/scbd/kronos-web/pull/43/files#diff-96e30be0e8959d65492c17e42aceffc8R177
please remove

File: plugins/global-component-registry.js [hunk 3]
```diff
@@ -1,10 +1,13 @@
 import Vue from 'vue';
 import { CBadge, CEmitRootEvent, CIcon } from '@coreui/vue';
+import AsyncComputed from 'vue-async-computed';
```
Comment (sbilodeau) [inline plugins/global-component-registry.js:3]: please remove
https://github.com/scbd/kronos-web/pull/43/files#diff-96e30be0e8959d65492c17e42aceffc8R177

File: plugins/global-component-registry.js [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
 
 const globalComponents = [ CBadge, CIcon ];
 const globalDirectives = [ CEmitRootEvent ];
 
 globalComponents.forEach((component) => Vue.component(component.name, component));
 globalDirectives.forEach((directive) => Vue.directive(directive.name, directive));
 
+Vue.use(AsyncComputed);
+
```
Comment (sbilodeau) [inline plugins/global-component-registry.js:12]: please remove
https://github.com/scbd/kronos-web/pull/43/files#diff-96e30be0e8959d65492c17e42aceffc8R177

File: components/ExportEmails.vue [hunk 5]
```diff
@@ -147,31 +147,37 @@ export default {
...
+    .map((rowEmails) => ({ emails: rowEmails.join(this.selectedSpearator) }));
 
-    this.rows = this.selectedRowSize ? 3 : 10;
-  }
+  this.rows = this.selectedRowSize ? 3 : 10;
 }
 
 function formatEmailAddress(emails) {
-  return emails.map((x) => (this.selectedFormat === 'NAME_EMAIL' ? `${x.name || ''}  <${x.address}>`.trim() : `${x.address}`.trim()));
+  return this.selectedFormat === 'NAME_EMAIL' ? `${emails.name || ''}  <${emails.address}>`.trim() : `${emails.address}`.trim();
```
[the remark this answers is not in the document]
Comment (JamesSCBD) [reply components/ExportEmails.vue]: Done!

File: components/MultiDeleteModal.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (JamesSCBD) [reply components/MultiDeleteModal.vue]: Done!

File: package.json (hunk 2 above)
Comment (JamesSCBD) [reply package.json]: Done!

File: components/MultiDeleteModal.vue [hunk 6]
```diff
@@ -170,13 +179,35 @@ export default {
...
 };
 
+async function items()    {
+  const skip       = this.perPage * (this.page);
+  const recordskip = this.selectedResult.totalRecordCount - skip;
+  const cursor     = this.selectedResult.getCursor({ skip: recordskip });
+  let obj          = null;
+  const list       = [];
+
+  while (obj = await cursor.next()) { // eslint-disable-line no-cond-assign, no-await-in-loop
```
Comment (sbilodeau) [inline components/MultiDeleteModal.vue:197]: you are still loading all the record after the offset.

if you have 1000 records and you skip 10... you will load 990 records.

Comment (JamesSCBD) [reply components/MultiDeleteModal.vue:197]: Then how can we do that? We need limit option too and i checked limit is not working

Comment (JamesSCBD) [reply components/MultiDeleteModal.vue:196]: `if (records.length >= this.perPage) break;`

I did same thing i forget this line that why my code not working. Thanks

Comment (sbilodeau) [reply components/MultiDeleteModal.vue:196]: yes :)
I also simplified the code
https://github.com/scbd/kronos-web/pull/43/commits/7cc39346bd9b599f8ecf868de7a4929bc05dd644

**Pull request merged**